pysindy/feature_library/base.py

```diff
@@ -525,40 +512,26 @@ def transform(self, x_full):
 
         xp_full = []
         for x in x_full:
-            # preallocate matrix
-            shape = np.array(x.shape)
-            shape[-1] = self.n_output_features_
-            xp = np.zeros(shape)
-
-            current_feat = 0
+            xp = []
             for i in range(len(self.libraries_)):
                 lib_i = self.libraries_[i]
-                lib_i_n_output_features = lib_i.n_output_features_
                 if self.inputs_per_library_ is None:
                     xp_i = lib_i.transform([x])[0]
                 else:
                     xp_i = lib_i.transform(
                         [x[..., np.unique(self.inputs_per_library_[i, :])]]
                     )[0]
+
                 for j in range(i + 1, len(self.libraries_)):
                     lib_j = self.libraries_[j]
-                    lib_j_n_output_features = lib_j.n_output_features_
                     xp_j = lib_j.transform(
                         [x[..., np.unique(self.inputs_per_library_[j, :])]]
                     )[0]
 
-                    start_feature_index = current_feat
-                    end_feature_index = (
-                        start_feature_index
-                        + lib_i_n_output_features * lib_j_n_output_features
-                    )
-
-                    xp[..., start_feature_index:end_feature_index] = self._combinations(
-                        xp_i, xp_j
-                    )
-
-                    current_feat += lib_i_n_output_features * lib_j_n_output_features
+                    for xp_ij in self._combinations(xp_i, xp_j):
+                        xp.append(xp_ij)
 
+            xp = np.array(xp)
```

Comment:
Is this supposed to be a concatenate?

Reply:
No, but it might be possible to better vectorize the products here. Currently, it loops through each pair of libraries, calculates the product of features (in a vectorized way), then appends each product to an increasing list `xp` of features. The final step here just converts the list back into a numpy axis cooresponding to `ax_coord`, as needed by `AxesArray`. Maybe there is a more general way to vectorize all products for an arbitrary list of terms...

Reply:
So like a `numpy.stack(axis=-1)`?